torch/distributed/device_mesh.py

```diff
@@ -327,8 +329,32 @@ def _setup_world_group_and_device(self):
             default_initialized = is_initialized()
             # TODO: think about how to allow pg options to be passed to world group
             # or mesh dimension groups
-            if not default_initialized:
+            use_torch_comms = int(os.environ.get("USE_TORCH_COMMS", 0))
+            if not default_initialized and not use_torch_comms:
                 init_process_group()
+            elif use_torch_comms:
+                backend = os.environ.get("TORCH_COMMS_BACKEND", "")
+                if not backend:
+                    raise RuntimeError("TORCH_COMMS_BACKEND is not set.")
+                import torchcomms
+                from torchcomms.device_mesh import (
+                    _create_torchcomm_process_group,
+                    _get_store_for_pg,
+                )
+
+                comm = torchcomms.new_comm(
+                    backend,
+                    torch.device(self._device_type),
+                    name="default_world_comm",
+                )
+                global_pg = _create_torchcomm_process_group(
```

Comment:
what is a torchcomm_process_group?  is this the c10d-like wrapper?  if so, why can't we make 'torchcomms' a backend type and rely on the orginal call to `init_process_group` under `not default_initialized` block above?

Reply:
Yes, this is the c10d wrapper. Intention is to move away from global PG apis which is why we're not just initializing a global PG as a backend and instead constructing it ourselves

Reply:
hmm. would it actually be materially worse for torchcomms if we just did use the init_process_group method here? i'd be interested in removing globalness from c10d and devicemesh over time if possible too, but it seems like for now that might be a lower-friction path to integrate